### ISSUE TITLE
polynomial type with non Eigen types

### ISSUE THREAD

**Issue by stonneau** [loco-3d/ndcurves#53]
In the current implementation of the polynomial class, the coefficients are stored using an Eigen matrix and not a vector.
If other types are used (such as linear_variable), the polynomial class will not work.
We should either change this or remove the point type form the template definition as it is useless.
The coefficients should be stored as std::vector.

Is there any reasons this has been done differently ?
@JasonChmn @proyan

**Comment by MaximilienNaveau** [comment]
Moreover you cannot create a simple 1D polynomial easily like this.
I was looking for a simple min jerk solver, I managed with ndcurves but it's a bit convoluted to use `Eigen::Vector::One(1)` for the initialization.

**Comment by stonneau** [comment]
Indeed. Do you have any suggestions on how to avoid that?

________________________________

**Comment by MaximilienNaveau** [comment]
Yes we should be implementing the template specialization for the type `double`.
The size of everything should be considered 1 or have an inner converter from double to Eigen and use the current implementation.
Not sure what is the best way... I am a very new user of ndcurves.

**Comment by stonneau** [comment]
We had that before, but the issue is with the python bindings I think. @JasonChmn  could you recall the issue
